File: java/src/main/lowestCommonAncestor/iterative.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        
        while((p.val -root.val) * (q.val - root.val) > 0){
            if(p.val > root.val){
                root = root.right;
            }
            else{
                root = root.left;
            }
        }
        return root; 
    }
}

File: python/99RecoverBinarySearchTree/iterative.py
class TreeNode:
    def __init__(self, val = 0, left = None, right = None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def recoverTree(self, root: Optional[TreeNode])->None:
        if not root:
            return
        stack = []
        first = second = pre = None
        while stack or root:
            if root:
                stack.append(root)
                root = root.left
            if pre and pre.val > root.val:
                second = root
                if not first:
                    pre = first
                else:
                    break
            pre = root
            root = root.right
        first.val, second.val = second.val, first.val

